boot.py

```diff
@@ -80,6 +80,11 @@ def _register_all_plugins() -> None:
     plugin_classes = []  # type: List[Type[AbstractPlugin]]
     _get_final_subclasses(AbstractPlugin.__subclasses__(), plugin_classes)
     for plugin_class in plugin_classes:
+        try:
+            if not plugin_class.name():
+                continue
+        except NotImplementedError:
+            continue
```

Comment:
Can't you check in _get_final_subclasses whether `d.__abstractmethods__` is not falsy?

Reply:
I don't think this would work.

NpmClientHandler doesn't have any abstract methods, same as a subclass of it like LspPyright.